src/components/MetricChart/MetricChart.tsx

```diff
@@ -112,7 +120,14 @@ export const MetricChart = ({
     width = 400,
     height = width / 1.5,
     chartOptions,
+    onChartDataStatusChange,
 }: DiagnosticsChartProps) => {
+    // YAGR charts use uplot inside
+    // uplot has the issue, that it doesn't render correctly inside not visible elements
+    // so if chart is used inside component with display:none, it will be empty, when visibility change
+    const containerRef = useRef<HTMLDivElement>(null);
+    const isChartVisible = useIsElementVisible(containerRef);
```

Comment:
It looks much more straightforward to pass `isChartVisible` property from parent.

Reply:
It's more straightforward, but it makes you to pass visible prop whenever you want to hide some component, that has charts inside. For example, if I'd want to add `NotRenderUntilFirstVisible` logic to tenant diagnostics tabs, it won't be possible without props drilling to ensure charts properly hidden and shown